### PR TITLE
add serializer and deserializer for `Signature`

### PR DESCRIPTION
**Pull request by mukulljangid**
addresses: https://github.com/XRPLF/xrpl4j/issues/304

### CODE REVIEW

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/SignatureSerializer.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+import java.io.IOException;
+
+/**
+ * Custom Jackson serializer for {@link Signature}es.
+ */
+class SignatureSerializer extends JsonSerializer<Signature> {
```
Comment (nkramer44) [inline xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/SignatureSerializer.java:12]: Let's move this and the deserializer into the same package as the other serializers/deserializers and add them to the `Xrpl4jModule` so you don't need the `using = SignatureSerializer` and `using = SignatureDeserializer.class`

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java [hunk 2]
```diff
@@ -12,8 +12,8 @@
  * Represents a digital signature for a transaction that can be submitted to the XRP Ledger.
  */
 @Value.Immutable
-@JsonSerialize(as = ImmutableSignature.class)
-@JsonDeserialize(as = ImmutableSignature.class)
+@JsonSerialize(using = SignatureSerializer.class)
+@JsonDeserialize(using = SignatureDeserializer.class)
```
Comment (nkramer44) [inline xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java:16]: These will probably get removed once you add the serializer/deserializer to the Xrpl4jModule, but for future reference, you need both the `as` and the `using` fields in these annotations for things to work correctly with immutables.

**Review by nkramer44 — CHANGES_REQUESTED** [review]

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java (hunk 2 above)
Comment (mukulljangid) [reply xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java:16]: https://github.com/XRPLF/xrpl4j/pull/303#discussion_r984657149

**Review by nkramer44 — APPROVED** [review]

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/SignatureSerializer.java (hunk 1 above)
Comment (sappenin) [reply xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/SignatureSerializer.java:12]: This still needs to be addressed, correct?

**Review by sappenin — CHANGES_REQUESTED** [review]

**Comment by mukulljangid** [comment]
> This still needs to be addressed, correct?

Not at the moment. That would need bigger refactor of modules

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/SignatureSerializer.java (hunk 1 above)
Comment (nkramer44) [reply xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/SignatureSerializer.java:12]: The problem is that `Signature` is in xrpl4j-crypto, which depends on `xrpl4j-model`, but all of the Jackson modules are in `xrpl4j-model`, so there's no way to move these without creating a circular dependency. We probably want to extract all the Jackson stuff into it's own module at some point, but I think that should be future work. WDYT?

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java (hunk 2 above)
Comment (sappenin) [reply xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java]: This will be fixed via #306.

**Review by sappenin — APPROVED** [review]

**Pull request merged**